packages/hardhat/contracts/v2/init/InitUniV2Router.sol

```diff
@@ -4,13 +4,15 @@ import {LibDiamond} from '../libraries/LibDiamond.sol';
 import {LibUniV2Router} from '../libraries/LibUniV2Router.sol';
 
 contract InitUniV2Router {
-  function init(address _uniswapV2Router02) public {
+  function init(address uniswapV2Router02, address uniswapV2Factory) public {
     LibUniV2Router.DiamondStorage storage s = LibUniV2Router.diamondStorage();
 
     if (!s.isInitialized) {
       s.isInitialized = true;
-      s.uniswapV2router02 = IUniswapV2Router02(_uniswapV2Router02);
+      s.uniswapV2router02 = IUniswapV2Router02(uniswapV2Router02);
       s.weth = IWETH(s.uniswapV2router02.WETH());
     }
+
+    s.uniswapV2Factory = uniswapV2Factory;
```

Comment:
Writing these is harmless, so the `isInitialized` isn't needed. For something like Kitty, it's very important not to overwrite the balances.